compiler/src/dotty/tools/dotc/transform/Splicer.scala

```diff
@@ -0,0 +1,35 @@
+package dotty.tools.dotc
+package transform
+
+import dotty.tools.dotc.core.Contexts._
+import dotty.tools.dotc.core.Symbols._
+import dotty.tools.dotc.ast.Trees._
+import dotty.tools.dotc.ast.tpd
+import dotty.tools.dotc.interpreter._
+import dotty.tools.dotc.quoted.PickledQuotes
+
+import scala.quoted
+
+/** Utility class to slice quoted expressions */
+object Splicer {
+  import tpd._
+
+  /** Splice the Tree for a Quoted expression. `~'(xyz)` becomes `xyz`
+   *  and for `~xyz` the tree of `xyz` is interpreted for which the
+   *  resulting expression is return as a `Tree`
+   */
+  def splice(tree: Tree)(implicit ctx: Context): Tree = tree match {
+    case Apply(quote, quoted :: Nil) if quote.symbol == defn.quoteMethod => quoted
```

Comment:
These seem to be used in a few separate places now. Make an extractor?

Reply:
Yes